src/realm/util/safe_int_ops.hpp

```diff
@@ -19,6 +19,11 @@
 #ifndef REALM_UTIL_SAFE_INT_OPS_HPP
 #define REALM_UTIL_SAFE_INT_OPS_HPP
 
+#ifdef _WIN32
+#undef max // collides with numeric_limits::max called later in this header file
+#undef min // collides with numeric_limits::min called later in this header file
+#endif
+
```

Comment:
Don't we also define the `NOMINMAX` macro, and shouldn't that serve this purpose as well?